omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx_chef_api_lb.conf.erb

```diff
@@ -45,8 +45,30 @@
       alias /opt/opscode/version-manifest.txt;
     }
 
-    <% if node['private_chef']['data_collector']['proxy'] -%>
+    <% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token']-%>
     location "/data-collector/" {
+      access_by_lua '
+        ngx.req.read_body()
+        local token = "<%= node['private_chef']['data_collector']['token'] %>"
+
+        if ngx.var.request_method ~= "POST"
+        then
+          ngx.exit(ngx.HTTP_NOT_FOUND)
+        end
+
+        -- TODO: auth with /validate_request endpoint
+
+        local h = ngx.req.get_headers()
+
+        if h["x-data-collector-token"] == nil
```

Comment:
Thinking about this, it's probably needed for backwards-compat (as @stevendanna pointed out [here](https://github.com/chef/chef-server/pull/984/files#diff-e61ee1a4529c1a04df73e9febe4bbe70R48)):

If the condition was changed to
```
<% if node['private_chef']['data_collector']['proxy'] || (node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token']) -%>
```
and the `TODO` block is only used when _the header is not already present_, this could work for both old setups and new ones. However, we'll ask for the trouble hinted at in my previous comment ^